### PR TITLE
fix(core): fix flaky durable providers

### PR DESCRIPTION
**Pull request by KamalAman**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #11664

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: integration/scopes/e2e/durable-providers.spec.ts [hunk 1]
```diff
@@ -84,6 +86,26 @@ describe('Durable providers', () => {
       );
       expect(result.body).deep.equal({ tenantId: '3' });
     });
+
+    it(`should not cache durable providers that throw errors`, async () => {
+      // How to test this?
+      // const constructorStub = sinon.stub(DurableService, 'constructor').throws(new Error('test error'));
```
[the remark this answers is not in the document]
Comment (micalevisk) [reply integration/scopes/e2e/durable-providers.spec.ts:92]: I don't know neither :/ I will try to find something later

**Review by micalevisk — CHANGES_REQUESTED** [review]
can you please remove the `yarn.lock`

**Comment by KamalAman** [comment]
i'll try to get back to this over the weekend and close it off

**Comment by dberardo-com** [comment]
@KamalAman @kamilmysliwiec  what was the reason why this PR was never accepted ? i reopened an issue on the matter: https://github.com/nestjs/nest/issues/13953

**Comment by KamalAman** [comment]
@dberardo-com yeah this is an annoying bug. I never figured out how to test it. Anyone who wants to pick it up, please feel free to take this over.

**Pull request closed without merging**